core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -1770,6 +1818,20 @@ void triggerListeningToNamespace(final MongoNamespace namespace) {
     }
   }
 
+  void triggerListeningToNamespace(
+      final MongoNamespace namespace, final BsonValue id, final boolean fromSync) {
+    syncLock.lock();
+    try {
+      if (instanceChangeStreamListener.isDocumentBeingWatched(namespace, id) == fromSync) {
```

Comment:
This variant is to make sure we only restart a listener if this is a new id being added.

Reply:
Nice, I like this change.